python/setup.py

```diff
@@ -7,6 +7,13 @@
 
 import glob
 import os
+import subprocess
+
+try:
+  subprocess.call(['ccache', '--version'])
+  os.environ['CC'] = 'ccache g++'
```

Comment:
People might compile with other c++ compilers - e.g. `clang` - just add an another `subprocess.call(["cc", "--version"])` and then `os.environ['CC'] = 'ccache c++'`